attestation-agent/attester/src/se/ibmse.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) Copyright IBM Corp. 2024
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Copyright (C) Copyright IBM Corp. 2024
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{anyhow, Result};
+use pv::{
+    request::BootHdrTags,
+    uv::{AttestationCmd, ConfigUid, UvDevice},
+};
+use serde::{Deserialize, Serialize};
+use serde_json;
+use serde_with::{base64::Base64, serde_as};
+
+#[allow(unreachable_code)]
+pub fn is_se_guest() -> bool {
+    #[cfg(not(target_arch = "s390x"))]
```

Comment:
Can we move this to attester/src/lib, since the whole module doesn't make sense if the target arch is not s390x?

Reply:
@mkulke there is a `#[cfg(feature = "se-attester")]` to build this module selectively, is that OK for you?

Reply:
to understand the reasoning behind this additional flag: the motivation is to compile the se-attester on all archs into the bin but disable the code at runtime on the wrong archs?

Reply:
Yes, intention is to use `#[cfg(feature = "se-attester")]` to include `se` or not, `#[cfg(not(target_arch = "s390x"))]` to make sure it works well also on x86 if `se-attester` selected.

Reply:
But is this what we want? As I understand se-attester would never work on x86_64, why package the code into the bins?

Reply:
Not sure whether its reasonable, I want it also be checked by CI before we add a s390x runner in CI and GHA.

Reply:
ok, to be fair we have the same for the arm attester, we'll include it in the x86_64 bins even if it's dead code. we can leave it like this, I suppose, but maybe check whether we can use pv_core instead to get rid of the openssl dep (it might cause issues in initrd-based guest-components)

Reply:
oh, I got your concern!

Reply:
Fixed, thanks!